### DISCUSSION TITLE
[1.20.1] NBT Filtering

### DISCUSSION THREAD

**Discussion by CYM-GS** [P3pp3rF1y/SophisticatedBackpacks#1077] [Feature / Idea Archive] [answered]
I really love NBT filtering to work, which would making filtering items like apotheosis rarities with `{affix_data:{rarity:"apotheosis:mythic"}}` much easier. Maybe making it a diamond or nether ite tier upgrade for the void and magnets upgrades.

**Comment by P3pp3rF1y** [comment, marked as answer]
This would be extremely complicated to code and probably even use in gui when setting up the filter or would require just having pretty much a textbox for player to enter nbt manually. 
The other part of it is that regular player probably has an understanding that items have an nbt, but likely no idea how to view nbt, what nbt on the item could be important, ... So this would really be an upgrade for very technical players.
And for these reasons as well as the fact that this seems to be going beyond what I would like to have in mod in terms of complexity I don't want to add it to the mod. This would be better suited for some pipe mod / filtering focused mod rather than a mod that's supposed to be primarily about storage and some additional functionality through upgrades.

**Reply by CYM-GS** [reply]
I understand your point and that this is a very niche feature that probably not many players of this mod will need.
There are a few pipe mods that already have this feature implemented and are also used by other players.
I was looking for a more mobile system than the pipe approach, but I can see now how much work that'd be and how few players will probably use it.
